woocommerce/woocommerce-android
# Improve Splash screen on Android 12

Closes: #5241

### Description
Android 12 forces Splash screens for all apps, and for cases where the Splash screen was just overriding the theme, the new implementation will override it.
Android 12 uses by default the launcher's icon by default, this PR adds the different attributes that Android 12 added to supply the correct background color and icon for the Splash screen.
The design is still a bit different from previous versions, since we can't supply an icon size, the system controls it:

#### Android 12
|   | Before | After |
| -- | -- | -- |
| Jalapeno | ![12_jalapeno_before](https://user-images.githubusercontent.com/1657201/141826482-3277aac6-8037-471d-a64e-94842cc01e65.png) | ![12_jalapeno_after](https://user-images.githubusercontent.com/1657201/141826527-a2855488-c666-48d8-9864-45bdaeef719e.png) |
| Wasabi | ![12_wasabi_before](https://user-images.githubusercontent.com/1657201/141826689-3cfcad75-bf85-4c63-a4ac-ba06b0ca37cb.png) | ![12_wasabi_after](https://user-images.githubusercontent.com/1657201/141826702-1a22b2d6-61b5-4a6b-9a2d-5f22db81b698.png) |
| Vanilla | ![12_vanilla_before](https://user-images.githubusercontent.com/1657201/141826832-2eee6c5a-f739-4d9a-9578-449a1d2a9057.png) | ![12_vanilla_after](https://user-images.githubusercontent.com/1657201/141826854-540d516f-31e2-4f91-94ed-bd63cb75192e.png) |

In the PR, I also tried to simplify the logic for the logos used in previous versions (more in the comments), and here is a table to confirm that there is no regression, except fixing the size of the icon in wasabi builds:

#### Android 11
|   | Before | After |
| -- | -- | -- |
| Jalapeno | ![11_jalapeno_before](https://user-images.githubusercontent.com/1657201/141827078-673845bd-0cd8-4c3e-bf41-92b624a48fa5.png) | ![11_jalapeno_after](https://user-images.githubusercontent.com/1657201/141827089-e6507e18-79b8-42ad-9a1a-4e74922ad0e1.png) |
| Wasabi | ![11_wasabi_before](https://user-images.githubusercontent.com/1657201/141827399-fe02c3d7-0f0a-4de2-8887-af9d4d408984.png) | ![11_wasabi_after](https://user-images.githubusercontent.com/1657201/141827420-5e563617-7f95-46ab-94e5-bd238b81206e.png) |
| Vanilla |![11_vanilla_before](https://user-images.githubusercontent.com/1657201/141827730-46a7c87c-ddc6-4e7d-801a-f92ab5004b95.png)|![11_vanilla_after](https://user-images.githubusercontent.com/1657201/141827659-f24cb3a4-6ce5-4da4-84aa-72e68a11e5a4.png)|


### Testing instructions
Test the different build flavors in Android 12 and confirm the Splash screen looks correct.

### Images/gif
